src/libasr/codegen/asr_to_llvm.cpp

```
@@ -6220,6 +6220,10 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     }
 
     void visit_Exit(const ASR::Exit_t &x) {
+        if (loop_or_block_end.empty()) {
+            throw CodeGenError("Error: 'exit' statement used outside of any loop or block.",
+                               x.base.base.loc);
+        }
```

Can we throw a semantic error in the front-end at ASR level?

---

The thing is asr is generated as well as ast when ran with those flags --show-asr and --show-ast but running without those is throwing this error and it is from derefernecing null pointer in asr_to_llvm.cpp so i added the like there.

---

@certik  @kmr-srbh  Can you give more information about what to be done.

---

Ok.